src/api/transactionSubmission/submit.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { submitTransaction } from "../../internal/transactionSubmission";
+import { AccountAuthenticator, AnyRawTransaction, FeePayerRawTransaction } from "../../transactions";
+import { PendingTransactionResponse } from "../../types";
+import { AptosConfig } from "../aptosConfig";
+
+/**
+ * A class to handle all `Submit` transaction operations
+ */
+export class Submit {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  transaction(args: {
+    transaction: AnyRawTransaction;
+    senderAuthenticator: AccountAuthenticator;
+    feePayerAuthenticator?: AccountAuthenticator;
+  }): Promise<PendingTransactionResponse> {
+    if (args.transaction instanceof FeePayerRawTransaction) {
+      if (!args.feePayerAuthenticator) {
+        throw new Error("You are submitting a Fee Payer transaction but missing the feePayerAuthenticator");
+      }
+    }
+    return submitTransaction({ aptosConfig: this.config, ...args });
+  }
+
+  multiAgentTransaction(args: {
+    transaction: AnyRawTransaction;
+    senderAuthenticator: AccountAuthenticator;
+    additionalSignersAuthenticators: Array<AccountAuthenticator>;
+    feePayerAuthenticator?: AccountAuthenticator;
+  }): Promise<PendingTransactionResponse> {
+    if (args.transaction instanceof FeePayerRawTransaction) {
+      if (!args.feePayerAuthenticator) {
+        throw new Error("You are submitting a Fee Payer transaction but missing the feePayerAuthenticator");
+      }
+    }
+    return submitTransaction({ aptosConfig: this.config, ...args });
+  }
```

Comment:
If it's a fee payer, we should also ensure that the FeePayerAddress is not 0x0 (this will always fail).  It should be the address of the actual fee payer on submission.

Reply:
we expect to have a `feePayerAuthenticator` that is the authenticator of the account who signed with `sign.transactionAsFeePayer` where we assign the signer address to the `transaction.feePayerAddress`